tests/backup/cluster_controller_test.go

```diff
@@ -0,0 +1,111 @@
+package tests
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	"github.com/portworx/torpedo/drivers/backup/controllers/cluster"
+	"github.com/portworx/torpedo/pkg/log"
+	. "github.com/portworx/torpedo/tests"
+)
+
+// This testcase demonstrates the various functionalities of the ClusterController
+var _ = Describe("{ClusterControllerDemo}", func() {
+	var (
+		testRailId           int
+		clusterControllerMap map[string]*cluster.ClusterController
+	)
+
+	JustBeforeEach(func() {
+		testRailId = 31313
+
+		Step("Add source-cluster controller to cluster-controller-map", func() {
+			// source-cluster controller will help manage the source cluster
+			err := cluster.AddSourceClusterControllerToMap(&clusterControllerMap, testRailId)
+			log.FailOnError(err, "failed to add source-cluster controller to cluster-controller-map")
+		})
+
+		Step("Add destination-cluster controller to cluster-controller-map", func() {
+			// destination-cluster controller will help manage the destination cluster
+			err := cluster.AddDestinationClusterControllerToMap(&clusterControllerMap, testRailId)
+			log.FailOnError(err, "failed to add destination-cluster controller to cluster-controller-map")
+		})
+	})
+
+	It("Cluster Controller Demo", func() {
```

Comment:
please add verifications to the test

Reply:
This test case serves as a demo, hopefully making it convenient for reviewers to understand the functionalities of the ClusterController. I will provide a link to a successful test run shortly.